relay-server/src/actors/project_cache.rs

```diff
@@ -333,14 +344,20 @@ impl ProjectSource {
         _redis: Option<RedisPool>,
     ) -> Self {
         let local_source = LocalProjectSourceService::new(config.clone()).start();
-        let upstream_source =
-            UpstreamProjectSourceService::new(config.clone(), upstream_relay).start();
+        let global_config = Arc::new(GlobalConfig::default());
```

Comment:
this empty global config will hopefully be overridden quickly, but i'd prefer we either don't start any services until we have the first globalconfig fetched from upstream, or we have some kinda static globalconfig that will be used at start-up

Reply:
We should not assume  anything here, and just do not process anything till the global config is in there. 
Instead of default here we might be able to use something like `enum` with different states, one is `init` or `empty` and another with fetched global config

Reply:
instead of dealing with an uninitialized variant in the service, wouldn't it be better to wait for the first instance of globalconfig in servicestate::start, if we are not gonna start relay without a fresh globalconfig anyway?

Reply:
We start Relay currently without project configs as well, and also should start the Relay without global config. 
Fetching can take very long time, or upstream currently unavailable and it's perfectly fine.  We just keep buffering for time being and once the configs refreshed  we continue processing.

Reply:
ok, i think i'll add it to the following PR, for this one i think its best we just check if the fetching and reading and writing works wrt globalconfigs